dsa-solutions/lc-solutions/0900-0999/0904-fruit-into-baskets.md

```diff
@@ -0,0 +1,200 @@
+                if basket[fruits[left]] == 0:
+                    del basket[fruits[left]]
+                left += 1
+            max_fruits = max(max_fruits, right - left + 1)
+        return max_fruits
+
+# Example usage:
+solution = Solution()
+print(solution.totalFruit([1,2,1]))  # Output: 3
+```
+
+#### Code (C++):
+
+```cpp
+#include <vector>
+#include <unordered_map>
+#include <algorithm>
+using namespace std;
+
+class Solution {
+public:
+    int totalFruit(vector<int>& fruits) {
+        unordered_map<int, int> basket;
+        int left = 0, max_fruits = 0;
+        for (int right = 0; right < fruits.size(); ++right) {
+            basket[fruits[right]]++;
+            while (basket.size() > 2) {
+                basket[fruits[left]]--;
+                if (basket[fruits[left]] == 0) {
+                    basket.erase(fruits[left]);
+                }
+                left++;
+            }
+            max_fruits = max(max_fruits, right - left + 1);
+        }
+        return max_fruits;
+    }
+};
+
+// Example usage:
+int main() {
+    Solution solution;
+    vector<int> fruits = {1, 2, 1};
+    int result = solution.totalFruit(fruits);  // Output: 3
+    return 0;
+}
+```
+
+#### Code (Java):
+
+```java
+import java.util.HashMap;
+import java.util.Map;
+
+class Solution {
+    public int totalFruit(int[] fruits) {
+        Map<Integer, Integer> basket = new HashMap<>();
+        int left = 0, maxFruits = 0;
+        for (int right = 0; right < fruits.length; ++right) {
+            basket.put(fruits[right], basket.getOrDefault(fruits[right], 0) + 1);
+            while (basket.size() > 2) {
+                basket.put(fruits[left], basket.get(fruits[left]) - 1);
+                if (basket.get(fruits[left]) == 0) {
+                    basket.remove(fruits[left]);
+                }
+                left++;
+            }
+            maxFruits = Math.max(maxFruits, right - left + 1);
+        }
+        return maxFruits;
+    }
+
+    public static void main(String[] args) {
+        Solution solution = new Solution();
+        int[] fruits = {1, 2, 1};
+        System.out.println(solution.totalFruit(fruits));  // Output: 3
+    }
+}
+```
+
+#### Code (JavaScript):
+
+```javascript
+var totalFruit = function (fruits) {
+  let basket = new Map();
+  let left = 0,
+    maxFruits = 0;
+  for (let right = 0; right < fruits.length; ++right) {
+    basket.set(fruits[right], (basket.get(fruits[right]) || 0) + 1);
+    while (basket.size > 2) {
+      basket.set(fruits[left], basket.get(fruits[left]) - 1);
+      if (basket.get(fruits[left]) === 0) {
+        basket.delete(fruits[left]);
+      }
+      left++;
+    }
+    maxFruits = Math.max(maxFruits, right - left + 1);
+  }
+  return maxFruits;
+};
+
+// Example usage:
+console.log(totalFruit([1, 2, 1])); // Output: 3
+```
+
+#### Complexity:
+
+- Time Complexity: \(O(n)\), where \(n\) is the length of the input array `fruits`. We traverse the array once with the two pointers.
```

Comment:
replace `\(O(n)\)` to `$O(n)$`, `\(n\)` to `$n$` and `Time Complexity:` to ` **Time Complexity:**`